File: fetch_movies/movies.py
import requests

BASE_URL = "https://jsonmock.hackerrank.com/api/moviesdata/search"

def get_movies(title):
    movies = []
    page = 1

    while True:
        try:
            response = requests.get(f"{BASE_URL}?page={page}&Title={title}")
            data = response.json().get('data', [])
            if not data:
                break
            movies.extend(data)
            page += 1
        except Exception as e:
            print("something happens when fetching data: ", str(e))
            break

    return movies

def get_title_movies(movies):
    title_movies = list(set(movie["Title"] for movie in movies))
    return sorted(title_movies)

def execute():
    title = input("")
    movies = get_movies(title)
    movies_names = get_title_movies(movies)
    print("\n".join(movies_names))

if __name__ == "__main__":
    execute()

File: fetch_movies/movies.js
const BASE_URL = "https://jsonmock.hackerrank.com/api/moviesdata/search";

const getMovies = async (title) => {
    let movies = [];
    let page = 1;

    while (true) {
        try {
            const response = await fetch(`${BASE_URL}?page=${page}&Title=${title}`);
            const data = await response.json();
            const moviesData = data.data || [];
            if (!moviesData.length) {
                break
            } else {
                movies.push(...moviesData);
                page++;
            }
        } catch (error) {
            console.log("Something happens when fetching data: ", error.message);
            break
        }
    }

    return movies
}

const getTitleMovies = (movies) => {
    const titleMovies = [...new Set(movies.map(movie => movie.Title))];
    return titleMovies.sort();
}

const execute = async () => {
    const title = await getUserInput();
    const movies = await getMovies(title);
    const moviesNames = getTitleMovies(movies);
    console.log(moviesNames.join("\n"));
}

const getUserInput = () => {
    return new Promise(resolve => {
        const readline = require('readline').createInterface({
            input: process.stdin,
            output: process.stdout
        });
        readline.question('', title => {
            readline.close();
            resolve(title);
        });
    });
}

if (require.main === module) {
    execute();
}